src/screen.rs

```diff
@@ -666,60 +665,16 @@ impl Screen {
         // Don't need to check for fish_wcwidth errors; this is done when setting up
         // omitted_newline_char in common.rs.
         let non_space_width = get_omitted_newline_width();
-        let term = term();
-        let term = term.as_ref();
         // We do `>` rather than `>=` because the code below might require one extra space.
         if screen_width > non_space_width {
-            let mut justgrey = true;
-            let add = |abandon_line_string: &mut WString, s: Option<CString>| {
-                let Some(s) = s else {
-                    return false;
-                };
-                abandon_line_string.push_utfstr(&str2wcstring(s.as_bytes()));
-                true
-            };
-            if let Some(enter_dim_mode) = term.and_then(|term| term.enter_dim_mode.as_ref()) {
-                if add(&mut abandon_line_string, Some(enter_dim_mode.clone())) {
```

Comment:
not to beat the dead horse too much, but it's pretty sad that our logic of

	if dim mode is supported, print dim mode,
	else if max_colors >= 238, print 237
	else ...

didn't work in practice because some terminals ignore dim even though
dim support is advertised (not sure if that's a bug in the terminfo
or a misunderstanding in our code)